daft/filesystem.py

```diff
@@ -174,38 +152,42 @@ def _resolve_paths_and_filesystem(
     # Canonical protocol shared by all paths.
     protocol = next(iter(canonicalized_protocols))
 
-    if filesystem is None:
-        # Try to get filesystem from protocol -> fs cache.
-        filesystem = _get_fs_from_cache(protocol)
-    elif isinstance(filesystem, fsspec.AbstractFileSystem):
-        # Wrap fsspec filesystems so they are valid pyarrow filesystems.
-        filesystem = PyFileSystem(FSSpecHandler(filesystem))
+    # Try to get filesystem from protocol -> fs cache.
+    cached_filesystem = _get_fs_from_cache(protocol, io_config)
 
     # Resolve path and filesystem for the first path.
     # We use this first resolved filesystem for validation on all other paths.
-    resolved_path, resolved_filesystem = _resolve_path_and_filesystem(paths[0], filesystem)
+    resolved_path, resolved_filesystem = _infer_filesystem(paths[0], io_config)
```

Comment:
Wouldn't this create a new filesystem from the io_config even if it is cached?

Reply:
Oh yes you're correct - fixed